src/Arcus.Scripting.ApiManagement/Arcus.Scripting.ApiManagement.psm1

```diff
@@ -1,3 +1,53 @@
+<#
+ .Synopsis
+  Backs up an API Management service.
+
+ .Description
+  The Backup-AzApiManagement cmdlet backs up an instance of an Azure API Management service by getting the account storage key and creating an new storage context. 
+  This cmdlet stores the backup as an Azure Storage blob.
+
+ .Parameter ResourceGroupName
+  The name of the of resource group under which the API Management deployment exists.
+
+ .Parameter StorageAccountName
+  The name of the Storage account for which this cmdlet gets keys.
+
+ .Parameter ServiceName
+  The name of the API Management deployment that this cmdlet backs up.
+
+ .Parameter TargetContainerName
+  The name of the container of the blob for the backup. If the container does not exist, this cmdlet creates it.
+
+ .Parameter TargetBlobName
+  The name of the blob for the backup. If the blob does not exist, this cmdlet creates it. 
+  This cmdlet generates a default value based on the following pattern: {Name}-{yyyy-MM-dd-HH-mm}.apimbackup
+
+ .Parameter PassThru
+  Indicates that this cmdlet returns the backed up PsApiManagement object, if the operation succeeds.
+
+ .Parameter DefaultProfile
+  The credentials, account, tenant, and subscription used for communication with azure.
+#>
+function Backup-AzApiManagementService {
+    param(
+        [string][parameter(Mandatory = $true)] $ResourceGroupName = $(throw "Resource group name is required"),
+        [string][parameter(Mandatory = $true)] $StorageAccountName = $(throw "Storage account name is required"),
+        [string][parameter(Mandatory = $true)] $ServiceName = $(throw "API managgement service name is required"),
+        [string][parameter(Mandatory = $true)] $TargetContainerName = $(throw "Name of the target blob container is required"),
+        [string][parameter(Mandatory = $false)] $TargetBlobName = $null,
+        [switch][parameter(Mandatory = $false)] $PassThru = $false,
+        [Microsoft.Azure.Commands.Common.Authentication.Abstractions.Core.IAzureContextContainer][parameter(Mandatory = $false)] $DefaultProfile = $null
```

Comment:
This is not being passed along to the actual implementation.
Do we require this?